include/trial/protocol/json/partial/skip.hpp

```diff
@@ -0,0 +1,91 @@
+#ifndef TRIAL_PROTOCOL_JSON_PARTIAL_SKIP_HPP
+#define TRIAL_PROTOCOL_JSON_PARTIAL_SKIP_HPP
+
+///////////////////////////////////////////////////////////////////////////////
+//
+// Copyright (C) 2018 Vinícius dos Santos Oliveira <vini.ipsmaker@gmail.com>
+//
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include <trial/protocol/json/reader.hpp>
+
+namespace trial
+{
+namespace protocol
+{
+namespace json
+{
+namespace partial
+{
+
+template<class CharT>
+typename basic_reader<CharT>::view_type
+skip(basic_reader<CharT> &reader, std::error_code &ec)
+{
+    using view_type = typename basic_reader<CharT>::view_type;
+    using size_type = typename view_type::size_type;
+
+    switch (reader.symbol()) {
+    case token::symbol::end:
```

Comment:
For `end` I would simply return an empty view without an error.

Reply:
Why? No tokens have been skipped. User fed the parser with wrong input. If it was unintentional, no silent pass should happen.

Reply:
I agree with you that skipping at the end is unintentional and should be raised as an error.

When I wrote my earlier comment, I wanted to align the behavior to `json::reader::next` which does allow calls at the end of the buffer. After further thinking I cannot see any good reasons for it. I tried to return an error instead, and all the unit-tests passed without a problem.

Advancing past the end with `std::advance` is undefined behavior. This also argues in favor of not accepting this behavior.